### PR TITLE
Added pull request creation and workflow suggestions.

### PR DESCRIPTION
**Pull request by matejak**
I have put together an update of the developer guide according to suggestions from [the mailing list](https://lists.fedorahosted.org/archives/list/scap-security-guide@lists.fedorahosted.org/thread/2ICQN6K5LA3RWLBSMVF7LT64OUGDQSHI/).
The goal is to make the development more transparent, and as a side-effect, more accessible.

### CODE REVIEW

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -73,11 +76,22 @@ image::Patching-Pull_Requests.png[align="left"]
...
+- assign yourself to the issue (if you have rights), or
+- mention the issue number in the pull request.
+
+This will improve the odds that multiple people won't work on a single issue without being aware of each other's work.
 After completing the form, select "Send pull request":
 
 image::Patching-Send_Pull_Request.png[align="left"]
 
-5) A community member will review your patch. They will either merge the patch upstream, indicate additional changes/documentation needed, or decline the patch. You'll automatically be notified via EMail of any status updates.
+5) Keep historical records clean - don't use git history-altering magic in your branch during your work on the pull request.
```
Comment (redhatrises) [inline docs/manual/developer_guide.adoc]: `Keep historical records clean`
What do you mean? Be verbose here.

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -73,11 +76,22 @@ image::Patching-Pull_Requests.png[align="left"]
...
+If you e.g. squash commits, the pull request page will be broken - if you made some mistakes, got feedback and corrected those mistakes based on the feedback, nobody will be able to learn from your pull request, because commits introducing mistakes will disappear and comments of reviewers therefore won't make sense.
+You can always squash commits while merging the pull request, or you can close a complicated pull request, create a new one (in another branch) with streamlined commits and reference the old PR in the streamlined one, so it is possible to backtrack what went on.
```
Comment (redhatrises) [inline docs/manual/developer_guide.adoc]: Squash unnecessary commits when merging the pull request, or close a complicated pull request and create a new one (in another branch) with streamlined commits. Reference the old PR in the new streamlined pull request so it is possible to backtrack what went on.

File: docs/manual/developer_guide.adoc [hunk 3]
```diff
@@ -73,11 +76,22 @@ image::Patching-Pull_Requests.png[align="left"]
...
+- mention the issue number in the pull request.
+
+This will improve the odds that multiple people won't work on a single issue without being aware of each other's work.
 After completing the form, select "Send pull request":
 
 image::Patching-Send_Pull_Request.png[align="left"]
 
-5) A community member will review your patch. They will either merge the patch upstream, indicate additional changes/documentation needed, or decline the patch. You'll automatically be notified via EMail of any status updates.
+5) Keep historical records clean - don't use git history-altering magic in your branch during your work on the pull request.
+If you e.g. squash commits, the pull request page will be broken - if you made some mistakes, got feedback and corrected those mistakes based on the feedback, nobody will be able to learn from your pull request, because commits introducing mistakes will disappear and comments of reviewers therefore won't make sense.
```
Comment (redhatrises) [inline docs/manual/developer_guide.adoc]: Sorry, no. All commits should be clean if changes are on the content that was reviewed e.g. https://github.com/OpenSCAP/scap-security-guide/pull/2409 or https://github.com/OpenSCAP/scap-security-guide/pull/2418 unless you are adding new functionality/fixes in a commit e.g. https://github.com/OpenSCAP/scap-security-guide/pull/2003. PRs with commits like https://github.com/OpenSCAP/scap-security-guide/pull/2280/commits are a no go as it dirties up the git history making it hard to determine what the actual change was (examples of others I just found this one easily)

File: docs/manual/developer_guide.adoc (hunk 1 above)
Comment (matejak) [reply docs/manual/developer_guide.adoc]: I have rephrased that, but I think that the following text elaborates what was meant by that.

**Comment by matejak** [comment]
@redhatrises Thank you very much for your help to make the PR more understandable!

File: docs/manual/developer_guide.adoc (hunk 3 above)
Comment (mpreisler) [reply docs/manual/developer_guide.adoc]: Squashing sometimes makes sense but IMO it should be an exception and I would prefer if developers did it via `git rebase -i` to maybe squash some neighboring commits or reorganize it to be more understandable.

I am against blanket squashing on merge as a rule, I think it makes it harder to understand the code. Sometimes I run git blame to figure out what some of the lines mean, I want to see the commit and the commit message, I don't want to run into a huge commit.

**Comment by mpreisler** [comment]
ACK. Great contribution!

**Pull request merged**